src/Learning/KWDataPreparation/KWClassStats.cpp

```diff
@@ -1177,28 +1197,29 @@ void KWClassStats::WriteJSONFields(JSONFile* fJSON)
 	// Parametrage eventuel de l'apprentissage supervise
 	if (GetTargetAttributeName() != "")
 	{
+		assert(not GetLearningSpec()->IsTargetStatsComputed() or
+		       GetTargetValueStats()->GetSourceAttributeNumber() == 0);
+
 		// Attribut cible
-		// On demande un affichage complet (source et cible) pour forcer
-		// l'utilisation explicite du libelle "Target"
-		assert(GetTargetValueStats()->GetSourceAttributeNumber() == 0);
 		fJSON->WriteKeyString("targetVariable", GetTargetAttributeName());
 
 		// Modalite cible principale
-		if (GetTargetAttributeType() == KWType::Symbol and GetMainTargetModalityIndex() != -1)
+		if (GetTargetAttributeType() == KWType::Symbol and GetMainTargetModality() != Symbol())
```

Comment:
I'm not sure I'm getting this: see the same comment above.

Reply:
Cf. commentaire plus haut